AsyncDisplayKit/ASPagerNode.m

```diff
@@ -84,6 +84,17 @@ - (ASCellNode *)collectionView:(ASCollectionView *)collectionView nodeForItemAtI
   return pageNode;
 }
 
+- (ASDataControllerCellNodeBlock)collectionView:(ASCollectionView *)collectionView nodeBlockAtIndexPath:(NSIndexPath *)indexPath {
+  ASDisplayNodeAssert(_pagerDataSource != nil, @"ASPagerNode must have a data source to load nodes to display");
+  if (![_pagerDataSource respondsToSelector:@selector(pagerNode:nodeBlockAtIndex:)]) {
```

Comment:
Although it feels like premature optimization at this stage, this method will be called hundreds or even thousands of times in a single run on the main thread for some applications, So we need to add a BOOL to cash weather the data source responds to the block selector.  Less problematic for the pager in some cases, but may as well do it for here, table, and collection.

Absolutely fine to do this in a follow-up diff, or even file a task on me to distribute it to the community member